misc/python/materialize/checks/all_checks/alter_connection.py

```diff
@@ -0,0 +1,324 @@
+from materialize.checks.checks import Check
+from materialize.checks.common import KAFKA_SCHEMA_WITH_SINGLE_STRING_FIELD
+
+
+def schema() -> str:
+    return dedent(KAFKA_SCHEMA_WITH_SINGLE_STRING_FIELD)
+
+
+class AlterConnectionToSsh(Check):
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            schema()
+            + dedent(
+                """
+                $[version>=5500] postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
+                ALTER SYSTEM SET enable_table_keys = true;
+                """
+            )
+            + dedent(
+                """
+                $ kafka-create-topic topic=alter-connection-1
+
+                $ kafka-ingest topic=alter-connection-1 format=bytes
+                one
+
+                > CREATE CONNECTION kafka_conn_alter_connection_1
+                  TO KAFKA (BROKER '${testdrive.kafka-addr}');
+
+                > CREATE SOURCE alter_connection_source_1
+                  FROM KAFKA CONNECTION kafka_conn_alter_connection_1 (TOPIC 'testdrive-alter-connection-1-${testdrive.seed}')
+                  FORMAT TEXT
+                  ENVELOPE NONE;
+
+                > SHOW CREATE CONNECTION kafka_conn_alter_connection_1;
+                materialize.public.kafka_conn_alter_connection_1 "CREATE CONNECTION \\"materialize\\".\\"public\\".\\"kafka_conn_alter_connection_1\\" TO KAFKA (BROKER = 'kafka:9092')"
+
+                > CREATE TABLE alter_connection_table (f1 INTEGER, f2 TEXT, PRIMARY KEY (f1));
+                > INSERT INTO alter_connection_table VALUES (1, 'abc');
+
+                > CREATE MATERIALIZED VIEW mv_alter_connection AS SELECT f1, f2 FROM alter_connection_table;
+
+                > CREATE SINK alter_connection_sink1 FROM mv_alter_connection
+                  INTO KAFKA CONNECTION kafka_conn_alter_connection_1 (TOPIC 'testdrive-alter-connection-sink-1-${testdrive.seed}')
+                  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_conn
+                  ENVELOPE DEBEZIUM;
+                """
+            )
+        )
+
+    def manipulate(self) -> list[Testdrive]:
+        return [
+            Testdrive(schema() + dedent(s))
+            for s in [
+                """
+                $ kafka-ingest topic=alter-connection-1 format=bytes
+                two
+
+                > ALTER CONNECTION kafka_conn_alter_connection_1 SET BROKER '${testdrive.kafka-addr}' USING SSH TUNNEL thancred;
+
+                $ kafka-ingest topic=alter-connection-1 format=bytes
+                three
+
+                $ kafka-create-topic topic=alter-connection-2
+
+                $ kafka-ingest topic=alter-connection-2 format=bytes
+                ten
+
+                > CREATE CONNECTION kafka_conn_alter_connection_2
+                  TO KAFKA (BROKER '${testdrive.kafka-addr}');
+
+                > CREATE SOURCE alter_connection_source_2
```

Comment:
CREATE SINK here too I would  say.

Reply:
Yeah, that's the plan. I wanted to check first if the current setup makes sense and then try to cleanup and stream-line it.